src/lib/eslint/EslintCommons.ts

```diff
@@ -0,0 +1,84 @@
+import { CLIEngine } from 'eslint';
+import * as path from 'path';
+import { CUSTOM_CONFIG } from '../../Constants';
+import { RuleResult, RuleViolation, ESMessage, ESRule, ESReport } from '../../types';
+import { FileHandler } from '../util/FileHandler';
+
+// Defining a function signature that will be returned by EslintStrategy.processRuleViolation()
+// This provides a safe way to pass around the callback function
+export interface ProcessRuleViolationType { (fileName: string, ruleViolation: RuleViolation): void}
```

Comment:
This interface defines the signature of the function that is returned by EslintStrategy's processRuleViolation(). Have a look at `addRuleResultsFromReport` and `toRuleResult` for the usage and the various implementations of processRuleViolation(), especially in TypescriptEslintStrategy and LWCEslintStrategy to get the full picture.